drivers/metal/rendering_device_driver_metal.mm

```diff
@@ -0,0 +1,3883 @@
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+/**************************************************************************/
+/*                                                                        */
+/* Portions of this code were derived from MoltenVK.                      */
+/*                                                                        */
+/* Copyright (c) 2015-2023 The Brenwill Workshop Ltd.                     */
+/* (http://www.brenwill.com)                                              */
+/*                                                                        */
+/* Licensed under the Apache License, Version 2.0 (the "License");        */
+/* you may not use this file except in compliance with the License.       */
+/* You may obtain a copy of the License at                                */
+/*                                                                        */
+/*     http://www.apache.org/licenses/LICENSE-2.0                         */
+/*                                                                        */
+/* Unless required by applicable law or agreed to in writing, software    */
+/* distributed under the License is distributed on an "AS IS" BASIS,      */
+/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or        */
+/* implied. See the License for the specific language governing           */
+/* permissions and limitations under the License.                         */
+/**************************************************************************/
+
+#include "rendering_device_driver_metal.h"
+
+#include "core/config/project_settings.h"
+#include "core/io/compression.h"
+#include "core/string/ustring.h"
+#include "pixel_formats.h"
+#include "rendering_context_driver_metal.h"
+
+#include "core/io/marshalls.h"
+#include "spirv_msl.hpp"
+#include "spirv_parser.hpp"
+#include <Metal/MTLTexture.h>
+#include <Metal/Metal.h>
+#import <compression.h>
```

Comment:
Aside from these, you might want to review the use of `#include` vs `#import`, which doesn't seem consistent across added `.mm` files.

```suggestion
#include "rendering_context_driver_metal.h"

#include "pixel_formats.h"

#include "core/config/project_settings.h"
#include "core/io/compression.h"
#include "core/io/marshalls.h"
#include "core/string/ustring.h"

#include <spirv_msl.hpp>
#include <spirv_parser.hpp>
#include <Metal/MTLTexture.h>
#include <Metal/Metal.h>
#import <compression.h>
```

Reply:
Yes, good catch. I'll change all the Objective-C++ to use `#import`, which is considered best practice.